moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1312,7 +1314,23 @@ void TrajectoryExecutionManager::executeThread(const ExecutionCompleteCallback&
 
   // only report that execution finished successfully when the robot actually stopped moving
   if (last_execution_status_ == moveit_controller_manager::ExecutionStatus::SUCCEEDED)
-    waitForRobotToStop(*trajectories_[i - 1]);
+  {
+    execution_trajectories_mutex_.lock();
+    if (!execution_complete_ && trajectories_.size() > 0)
```

Comment:
Should this check that the trajectory size is greater than or equal to `i` instead?